diff --git a/Makefile b/Makefile
index afa13c2..0474860 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,10 @@
-PREFIXDIR = /usr/local
-BINDIR = /bin
-MANDIR = /usr/share/man/man8
+prefix ?= /usr/local
+bindir ?= /bin
+mandir ?= /usr/share/man/man8
 PROG = numatop
-CC = gcc
-LD = gcc
-CFLAGS = -g -Wall -O2
+CC ?= gcc
+LD ?= gcc
+CFLAGS ?= -g -Wall -O2
 LDLIBS = -lncurses -lpthread -lnuma
 
 COMMON_OBJS = cmd.o disp.o lwp.o node.o numatop.o page.o perf.o \
@@ -16,7 +16,7 @@ INTEL_OBJS = wsm.o snb.o nhm.o
 all: $(PROG)
 
 $(PROG): $(COMMON_OBJS) $(INTEL_OBJS)
-	$(LD) $(LDLIBS) -o $@ $(COMMON_OBJS) $(INTEL_OBJS)
+	$(LD) $(LDFLAGS) -o $@ $(COMMON_OBJS) $(INTEL_OBJS) $(LDLIBS)
 
 %.o: ./common/%.c ./common/include/*.h
 	$(CC) $(CFLAGS) -o $@ -c $<
@@ -25,9 +25,9 @@ $(PROG): $(COMMON_OBJS) $(INTEL_OBJS)
 	$(CC) $(CFLAGS) -o $@ -c $<
 
 install: $(PROG)
-	install -m 0755 $(PROG) $(PREFIXDIR)$(BINDIR)/
+	install -D -m 0755 $(PROG) $(DESTDIR)$(prefix)$(bindir)/$(PROG)
 	gzip -c numatop.8 > numatop.8.gz
-	mv -f numatop.8.gz $(MANDIR)/
+	install -D numatop.8.gz $(DESTDIR)$(mandir)/numatop.8.gz
 
 clean:
 	rm -rf *.o $(PROG)
